containers/FishCaught.js

```diff
@@ -7,21 +7,93 @@ import {
   FlatList,
   TouchableOpacity,
 } from "react-native";
+import { fetchFish } from "../ApiCalls";
+import { setFish } from "../actions";
+import { connect } from "react-redux";
+import Fish from "../components/Fish";
 
-export default function FishCaught ({ navigation }) {
-  return (
-    <View>
-      <TouchableOpacity
-        style={styles.button}
-        onPress={() => {
-          navigation.navigate("AddFish");
-        }}
-      >
-        <Text style={styles.buttonText}>Add Fish</Text>
-      </TouchableOpacity>
-    </View>
-  );
-};
+class FishCaught extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentFish: [],
+      isLoading: true,
+    };
+  }
+
+  async componentDidMount() {
+    fetchFish()
+      .then((data) => this.props.setFish(data.data))
+      .then((fish) =>
+        this.setState({ currentFish: fish.data, isLoading: false })
+      );
+  }
+
+  async componentDidUpdate() {
+    if (this.state.currentFish.length != this.props.currentFish.length) {
+      fetchFish()
+        .then((data) => this.props.setFish(data.data))
+        .then((fish) => this.setState({ currentFish: fish.data }))
+    }
+  }
+
+  checkFish = () => {
+    if (this.props.currentFish.length > 0 && !this.state.isLoading) {
+      return (
+        <FlatList
+          data={this.props.currentFish}
+          renderItem={({ item }) => (
+            <Fish fish={item} navigation={this.props.navigation} />
+          )}
+        />
+      );
+    } else if (this.state.isLoading) {
+      return (
+        <View
+          style={{
+            alignItems: "center",
+            justifyContent: "center",
+            paddingTop: 50,
+          }}
+        >
+          <Text style={{ fontSize: 30, fontWeight: "bold" }}>
+            Loading Fish Data...
```

Comment:
We can add in a spinner or something more exciting then this message if we'd like. Wanted to at least get this added for a more clear UX